src/monattack.cpp

```diff
@@ -257,9 +257,8 @@ void mattack::shriek_stun(monster *z, int index)
     }
 
     Creature *target = z->attack_target();
-    int dist;
-    if( target == nullptr || (dist = rl_dist( z->pos(), target->pos() )) > 7 ||
-        !z->sees( *target ) ) {
+    int dist = rl_dist( z->pos(), target->pos() );
```

Comment:
`target` could be null. Simplest solution is to just initialized `dist` to 0 and keep the original `if` condition. Btw. gcc 5.2.0 also complains about this, but clang 3.7.0 doesn't.